File: Semana11/Dia3/sgp-construccion-back/app/models/Categoria.ts
import { Sequelize, DataTypes } from 'sequelize';

export const categoria_model: Function = ( conexion: Sequelize ) => {
       const modelo = conexion.define('Categoria',{
             cat_id: {type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true, allowNull: false},
             cat_nom: {type: DataTypes.STRING(45), allowNull: false},
       },{
             tableName: 't_categoria',
             timestamps: true              
       });
       return modelo             
}

File: Backend/Semana8/Dia2/models/categoria.py
from db import db

class CategoriaModel(db.Model):
    __tablename__ = "t_categoria"
    # el autoincremento en nuestro es por defecto
    #  the primary_key by default is auto_increment
    id = db.Column("cat_id",db.Integer,primary_key=True)
    descripcion = db.Column("cat_desc",db.String(30))
    # Relaciones
    productos = db.relationship('ProductoModel',lazy="dynamic")
    # constructor
    def __init__(self,desc):
        self.descripcion = desc
    
    def json(self):
        return {
            'id': self.id,
            'description': self.descripcion
        }

    def guardar_en_bd(self):
        db.session.add(self)
        db.session.commit()